src/vstest.console/CommandLine/InferHelper.cs

```diff
@@ -133,35 +133,24 @@ public Architecture AutoDetectArchitecture(IList<string?>? sources, Architecture
     public Framework AutoDetectFramework(IList<string?>? sources, out IDictionary<string, Framework> sourceToFrameworkMap)
     {
         sourceToFrameworkMap = new Dictionary<string, Framework>();
-        Framework framework = Framework.DefaultFramework;
 
         if (sources == null || sources.Count == 0)
-            return framework;
+            return Framework.DefaultFramework;
 
-        try
+        var framework = DetermineFramework(sources, out sourceToFrameworkMap, out var conflictInFxIdentifier);
+        if (conflictInFxIdentifier)
         {
-            var finalFx = DetermineFrameworkName(sources, out sourceToFrameworkMap, out var conflictInFxIdentifier);
-            TPDebug.Assert(finalFx != null, "finalFx should not be null");
-            framework = Framework.FromString(finalFx.FullName);
-            if (conflictInFxIdentifier)
-            {
-                // TODO Log to console and client.
-                EqtTrace.Info(
-                    "conflicts in Framework identifier of provided sources(test assemblies), using default framework:{0}",
-                    framework);
-            }
-        }
-        catch (Exception ex)
```

Comment:
Try/catch was only catching NRE when `finalFx` is `null` which happens only when all sources provided are `null`. `DetermineFrameworkName` and `Framework.FromString` are having inner try/catch.